### DISCUSSION TITLE
Changing username case

### DISCUSSION THREAD

**Discussion by mechalynx** [community/community#22822] [New to GitHub]
<p>I want to change my username to all lowercase but it’s unclear from:</p>
<p><a href="https://help.github.com/articles/what-happens-when-i-change-my-username/" rel="nofollow noopener">https://help.github.com/articles/what-happens-when-i-change-my-username/</a></p>
<p>whether changing a username’s case alone will have any side-effects. My guess is that it won’t but I thought I’d ask just to be sure and I want to request that this particular detail is made clear in the official FAQ.</p>
<p>Thanks.</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p><a class="mention" href="/mechalynx">@mechalynx</a> You are correct that case doesn’t effect your username. I will definitely share with the appropriate team your feedback on making this more clear in our documentation. </p>
<p>Cheers!</p>

**Comment by mechalynx** [comment]
<p>Thankyou! :smileyvery-happy:</p>

**Comment by majormoses** [comment]
<p>That seems to only be true for the majority of features but not all. For example git protocol operations (and possibly others) it will not handle it correctly, see <a href="https://github.com/github/hub/issues/157#issuecomment-4857834" rel="nofollow noopener">https://github.com/github/hub/issues/157#issuecomment-4857834</a> for an example. There are a handful of edgecases and they probably only affect some workflows but I don’t know that I would say it has 0 impact aside from the visual effects. Due to the lack of clarity around what and is not effected there have been a handful of issues (and will continue to be) with various tooling that do not handle this properly. Not trying to make anyone scared about renaming it for their own personal repos. For large OSS or proprietary  projects it should be done after additional considerations for how this affects your users workflows.</p>

**Comment by mechalynx** [comment]
<p>Thanks for this, at the very least there’s a linked example so people who might want to change username case can be aware before they do so.</p>

**Comment by PedroBern** [comment]
<p>I’ve tried to change the case of my username, to make it all lowercase, but I get the following message:</p>
<pre><code>Username pedrobern is not available. Please choose another. To submit a trademark claim, please see our Trademark Policy.
</code></pre>
<p>This dosen’t make sense because there is no one with this username besides me.</p>
<p>How should I proceed?</p>

**Comment by aabouzaid** [comment]
<p>I’ve the same issue as PedroBern<br>
I cannot change my account name to make it lowercase.</p>
<p>I get the same message.</p>

**Comment by PedroBern** [comment]
<p><a class="mention" href="/nadiajoyce">@nadiajoyce</a> could you help me please?</p>

**Comment by nyahbhinghiPrincess** [comment]
<p>Hello <a class="mention" href="/pedrobern">@pedrobern</a> and <a class="mention" href="/aabouzaid">@aabouzaid</a> if you haven’t already, please contact our <a href="https://github.com/contact" rel="nofollow noopener">private support</a> with the info so we can look into that further.</p>

**Comment by mfaisalhyder** [comment]
<p>Hi,</p>
<p>It feels good to see others are having the same issue as yourself :stuck_out_tongue:</p>
<p>Can i also contact support to get my username changed from MFaisalHyder -&gt; mfaisalhyder</p>

**Reply by JinxSeven** [reply]
Hello, I see that you’ve changed your username now
Im planning on doing the same thing any side effects?

**Comment by emberquill** [comment]
<p>I was able to get around the case-insensitive issue by changing my username twice. I started as EmberQuill, then changed my name to emberquill1, then emberquill.</p>
<p>There’s a minor risk of someone claiming your name during the brief time between the two changes, but the odds of that actually happening are almost zero. My username was only available to others for a few seconds.</p>

**Comment by kneasle** [comment]
<p>I’ve just tried to follow <a class="mention" href="/emberquill">@emberquill</a>’s example, and it turns out that I can’t change my username back to <code>kneasle</code> because my (plentiful) repositories have all been archived and their names reserved.  I thought I’d share this experience for others trying to do this.</p>
<p>Knowing that this can happen, I would go to GitHub help straight away and just ask them to perform the entire renaming operation for you - that way there’s no risk of getting your account stuck as something silly.</p>

**Comment by kneasle** [comment]
<p>Quick update - after successfully changing my username back with the help of github support, I discovered why I had had issues when <a class="mention" href="/emberquill">@emberquill</a> had not.</p>
<p>It turns out that if a repository has enough git clones in a week (100 or more, to be precise) then it is considered popular enough to have its URL reserved (archived, I think the term is). This means that if any of your repositories have their github urls reserved, you will (like I did) be unable to remane your username back to its original name without contacting Github support because doing so would overwrite and archived url. Admittedly it will overwrite it with a perfect copy of what was there before, but github seems to take a better-safe-than-sorry approach.</p>
<p>I would also like to point out that Github support were extremely fast in responding and then fixing my problem. &lt;3</p>

**Comment by kent-h** [comment]
FYI: It appears to now be possible to change the case of your username without changing to some other name first.

**Comment by mfaisalhyder** [comment]
Hey, nothing, everything works fine.
